cwltool/provenance.py

```diff
@@ -324,13 +322,13 @@ def __init__(self,
         self.user_provenance = user_provenance
         self.engine_uuid = research_object.engine_uuid
         self.add_to_manifest = self.research_object.add_to_manifest
-        if self.orcid:
+        if self.orcid is not None:
             _logger.debug(u"[provenance] Creator ORCID: %s", self.orcid)
         self.full_name = full_name or None
-        if self.full_name:
+        if self.full_name is not None:
             _logger.debug(u"[provenance] Creator Full name: %s",
                           self.full_name)
-        if not run_uuid:
+        if run_uuid is None:
```

Comment:
An empty string as `run_uuid` will (should!) cause failure as it is not a valid UUID.